dask/dataframe/tests/test_dataframe.py

```diff
@@ -3160,6 +3166,24 @@ def test_apply():
         ddf.apply(lambda xy: xy, axis="index")
 
 
+@pytest.mark.parametrize("convert_dtype", [True, False])
+def test_apply_convert_dtype(convert_dtype):
+    """Make sure that explicit convert_dtype raises a warning with pandas>=2.1"""
+    df = pd.DataFrame({"x": [1, 2, 3, 4], "y": [10, 20, 30, 40]})
+    ddf = dd.from_pandas(df, npartitions=2)
+    if PANDAS_GT_210:
+        ctx = pytest.warns(FutureWarning, match="the convert_dtype parameter")
+        with ctx:
+            expected = df.x.apply(lambda x: x + 1, convert_dtype=convert_dtype)
+        with ctx:
+            result = ddf.x.apply(lambda x: x + 1, convert_dtype=convert_dtype)
+    else:
+        expected = df.x.apply(lambda x: x + 1, convert_dtype=convert_dtype)
+        with pytest.warns(UserWarning, match="You did not provide metadata"):
```

Comment:
This is unrelated to the `convert_dtype` deprecation we're primarily checking here. Let's just explicitly provide a `meta` so we don't need to worry about the (unrelated) meta warning